modules/azure-auth-header/azure-auth-header.c

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (c) 2020 One Identity
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "azure-auth-header.h"
+#include "driver.h"
+#include "modules/http/http-signals.h"
+#include "messages.h"
+#include "compat/openssl_support.h"
+
+#define AZURE_AUTH_HEADER_PLUGIN "azure-auth-header"
+
+#define MAX_DATE_LEN 64
+
+struct _AzureAuthHeaderPlugin
+{
+  LogDriverPlugin super;
+
+  gsize   secret_len;
+  guchar *secret;
+
+  gchar   *workspace_id;
+  gchar   *method;
+  gchar   *path;
+  gchar   *content_type;
+};
+
+static gsize
+_get_rfc1123date(gchar *buf, gsize buf_len)
+{
+  time_t now = time(NULL);
+  struct tm gmt;
+
+  gmtime_r(&now, &gmt);
+
+  gchar format[] = "%a, %d %b %Y %H:%M:%S GMT";
+  gsize len = strftime(buf, buf_len, format, &gmt);
+
+  return len;
+}
+
+static void
+_get_rfc1123data_or_assert(gchar *date)
+{
+  gsize date_len = 0;
+  date_len = _get_rfc1123date(date, MAX_DATE_LEN);
+  g_assert(date_len);
+}
+
+static GString *
+_azure_auth_header_get_str_to_hash(AzureAuthHeaderPlugin *self, glong content_len, const gchar *date)
+{
+  gchar *fmt = "%s\n%ld\n%s\nx-ms-date:%s\n%s";
```

Comment:
This should be const.

Reply:
...or `static const`  (or just a string literal defined by a macro...)